Assets/Scripts/PlayerController.cs

```diff
@@ -243,50 +243,40 @@ private void cancelJump()
 
   private void interact()
   {
+    bool shouldInteract = interactAction != null && interactAction.triggered && isInteractPressed;
+
+    if (!shouldInteract) return;
 
-    if (interactAction != null && interactAction.triggered && isInteractPressed)
+    if (interactable != null)
     {
-      // Todo
-      // Need more conditionals here
-      // Handle appropriate interactions & animations
-      // Limit interaction options to the ones listed in the object being interacted with
-      if (interactedObject != null)
-      {
-        interactableController.HandleInteraction();
-      }
-      else
-      {
-        // handle non-object scene interactions
-        // like sitting or meowing
-      }
+      UseObjectInteraction(interactAction.name);
+      return;
     }
-  }
 
-  private void onUsePaw() { Debug.Log("Paw"); }
-  private void onUseMouth() { Debug.Log("Mouth"); }
-  private void onUseBody() { Debug.Log("Body"); }
+    UseDefaultReaction(interactAction.name);
+  }
 
   // To-do: This should probably be updated to a method that tests proximity to an interactable
   void OnTriggerEnter(Collider other)
   {
-    interactedObject = other.gameObject;
+    GameObject interactedObject = other.gameObject;
+
     if (interactedObject.CompareTag("Interactable"))
     {
       // Set our script reference to our newly collided interactable gameobject
-      interactableController = interactedObject.GetComponent(typeof(InteractableController)) as InteractableController;
-      interactionUIController = interactedObject.GetComponent(typeof(InteractionUIController)) as InteractionUIController;
-
-      // Show the interaction tool tip
-      interactionUIController.ShowInteractionTip(true);
+      interactable = interactedObject.GetComponent<Interactable>();
```

Comment:
`interactable` seems to be the only reference we need for the interaction system now 👌